cli/fathom_web/commands/vectorize.py

```diff
@@ -0,0 +1,312 @@
+    shutdown.
+    """
+    print('Starting HTTP file server...', end='', flush=True)
+    RequestHandler = partial(SilentRequestHandler, directory=samples_directory)
+    server = HTTPServer(('localhost', 8000), RequestHandler)
+    server_thread = Thread(target=server.serve_forever)
+    server_thread.start()
+    print('Done')
+    return server, server_thread
+
+
+def configure_firefox(fathom_fox, fathom_trainees, output_directory, show_browser, temp_dir):
+    """
+    Configures and launches Firefox to run the vectorizer with.
+
+    Sets headless mode, sets the download directory to the desired output
+    directory, turns off page caching, and installs FathomFox and Fathom
+    Trainees.
+
+    We return the webdriver object, and the process IDs for both the Firefox
+    process and the geckoview driver process so we can shutdown either
+    gracefully or ungracefully.
+    """
+    print('Configuring Firefox...', end='', flush=True)
+    options = webdriver.FirefoxOptions()
+    options.headless = not show_browser
+
+    profile = webdriver.FirefoxProfile()
+    profile.set_preference('browser.download.folderList', 2)
+    profile.set_preference('browser.download.dir', str(pathlib.Path(output_directory).absolute()))
+    profile.set_preference('browser.cache.disk.enable', False)
+    profile.set_preference('browser.cache.memory.enable', False)
+    profile.set_preference('browser.cache.offline.enable', False)
+
+    firefox = webdriver.Firefox(
+        options=options,
+        firefox_profile=profile,
+        service_log_path=f'{temp_dir.name}/geckodriver.log',
+    )
+
+    firefox.install_addon(fathom_fox, temporary=True)
+    firefox.install_addon(fathom_trainees, temporary=True)
+    print('Done')
+    return firefox, firefox.capabilities['moz:processID'], firefox.service.process.pid
+
+
+def run_vectorizer(firefox, sample_filenames):
+    """
+    Set up the vectorizer and run it, creating the vectors file.
+
+    Navigate to the vectorizer page of FathomFox, paste the sample filenames
+    into the text area and hit the vectorize button.
+
+    We monitor the status text area for errors and to see how many samples
+    have been vectorized, so we know when the vectorizer has stopped running.
+    """
+    print('Configuring Vectorizer...', end='', flush=True)
+    # Navigate to the vectorizer page
+    fathom_fox_uuid = get_fathom_fox_uuid(firefox)
+    firefox.get(f'moz-extension://{fathom_fox_uuid}/pages/vector.html')
+
+    pages_text_area = firefox.find_element_by_id('pages')
+    pages_text_area.send_keys(sample_filenames)
+
+    downloads_dir = pathlib.Path(firefox.profile.default_preferences['browser.download.dir'])
+    vector_files_before = set(downloads_dir.glob('vector*.json'))
+    number_of_samples = len(sample_filenames.splitlines())
+    status_box = firefox.find_element_by_id('status')
+    vectorize_button = firefox.find_element_by_id('freeze')
+    completed_samples = 0
```

Comment:
We should also provide an option to select which `type` to train on (for example, the Fathom Smoot articles ruleset has both a `"paragraph"` and `"article"` type). Currently this will just vectorize whichever type is default selected in the dropdown.

Reply:
I agree. I hadn't thought about that. I'll add that for this PR.